common/types/db-types.ts

```diff
@@ -0,0 +1,39 @@
+export type DetailedRating = {
+  readonly location: number;
+  readonly safety: number;
+  readonly value: number;
+  readonly maintenance: number;
+  readonly communication: number;
+  readonly conditions: number;
+};
+
+export type Review = {
+  readonly id: string;
+  readonly aptId: string | null;
+  readonly landlordId: string;
+  readonly overallRating: number;
+  readonly detailedRatings: DetailedRating;
+  readonly reviewText: string;
+  readonly date: Date;
+};
+
+export type Landlord = {
+  readonly id: string;
+  readonly name: string;
+  readonly contact: string | null;
+  readonly avgRating: number;
+  readonly photos: readonly string[]; // can be empty
+  readonly reviews: readonly string[]; // array of Review IDs in reviews collection
+  readonly properties: readonly string[]; // array of Apartment IDs in apartments collection
+};
+
+export type Apartment = {
+  readonly id: string;
+  readonly name: string;
+  readonly address: string; // may change to placeID for Google Maps integration
+  readonly landlordId: string | null;
+  readonly numBaths: number | null;
+  readonly numBeds: number | null;
+  readonly photos: string[]; // can be empty
```

Comment:
should this be readonly?

Reply:
yeah, the idea is that when you fetch stuff from the DB you shouldn't be mutating it anyway since it won't actually cause any changes to the data. If we need to alter any fields, we can just use spreading syntax to create a new immutable copy.

Reply:
Oh I meant `readonly photos: readonly string[]`

Reply:
If we don't add the second readonly we will still be able to mutate it which isn't what we want

Reply:
oops i forgot to add it there, will do it now